tests/testthat/test-preparedata-scatterplot.R

```diff
@@ -149,3 +149,152 @@ test_that("Handle y-values in multiple columns + multiple statistics",
+    "$120,001 to $150,000", "$150,001 to $200,000", "$200,001 or more",
+    "NET", "Male", "Female", "NET")), statistic = "%", basedescriptiontext = "sample size = from 707 to 800; total sample size = 800; 93 missing", basedescription = list(
+    Minimum = 707L, Maximum = 800L, Range = TRUE, Total = 800L,
+    Missing = 93L, EffectiveSampleSize = 800L, EffectiveSampleSizeProportion = 100,
+    FilteredProportion = 0), questiontypes = "PickAny", span = list(
+    rows = structure(list(c("Income", "Income", "Income", "Income",
+    "Income", "Income", "Income", "Income", "Income", "Income",
+    "Gender", "Gender", "Gender"), c("Less than $15,000", "$15,001 to $30,000",
+    "$30,001 to $45,000", "$45,001 to $60,000", "$60,001 to $90,000",
+    "$90,001 to $120,000", "$120,001 to $150,000", "$150,001 to $200,000",
+    "$200,001 or more", "NET", "Male", "Female", "NET")), class = "data.frame", .Names = c("",
+    ""), row.names = c(NA, 13L))), name = "table.BANNER", questions = c("BANNER",
+"SUMMARY")), Z2 = NULL, groups = NULL, labels = NULL, structure(c(`Less than $15,000` = -15.85213900942,
+`$15,001 to $30,000` = -11.4519011349191, `$30,001 to $45,000` = -11.7778446811784,
+`$45,001 to $60,000` = -7.2146350335478, `$60,001 to $90,000` = -4.85154432316768,
+`$90,001 to $120,000` = -9.33326808423343, `$120,001 to $150,000` = -12.918647093086,
+`$150,001 to $200,000` = -14.1409353915585, `$200,001 or more` = -14.385393051253,
+NET = 39.884207400925, Male = 11.4020968466331, Female = 12.1681291929185,
+NET = 42.4264068711928), .Dim = 13L, .Dimnames = list(c("Less than $15,000",
+"$15,001 to $30,000", "$30,001 to $45,000", "$45,001 to $60,000",
+"$60,001 to $90,000", "$90,001 to $120,000", "$120,001 to $150,000",
+"$150,001 to $200,000", "$200,001 or more", "NET", "Male", "Female",
+"NET")), statistic = "z-Statistic", basedescriptiontext = "sample size = from 707 to 800; total sample size = 800; 93 missing", basedescription = list(
+    Minimum = 707L, Maximum = 800L, Range = TRUE, Total = 800L,
+    Missing = 93L, EffectiveSampleSize = 800L, EffectiveSampleSizeProportion = 100,
+    FilteredProportion = 0), questiontypes = "PickAny", span = list(
+    rows = structure(list(c("Income", "Income", "Income", "Income",
+    "Income", "Income", "Income", "Income", "Income", "Income",
+    "Gender", "Gender", "Gender"), c("Less than $15,000", "$15,001 to $30,000",
+    "$30,001 to $45,000", "$45,001 to $60,000", "$60,001 to $90,000",
+    "$90,001 to $120,000", "$120,001 to $150,000", "$150,001 to $200,000",
+    "$200,001 or more", "NET", "Male", "Female", "NET")), class = "data.frame", .Names = c("",
+    ""), row.names = c(NA, 13L))), name = "table.BANNER.3", questions = c("BANNER",
+"SUMMARY")))
+
+test_that("Using tables with banners",
+{
+    tables.with.banners[[7]] <- PrepareForCbind(tables.with.banners[[7]])
+    expect_warning(pd <- PrepareData("Scatter", input.data.raw = tables.with.banners), "discarded")
```

Comment:
There were two warnings in this code here. Could use capture_warnings and test for both? Here the first warning stated that only the first column of X is used which is probably the correct one to capture? The second warning is the one that seems captured in this test about discarding row names. This seems to be due to the NET row appearing twice and the second one being removed, causing a dimension to change and warning thrown.

Reply:
I've made a change in 330a9b2 (around line 320) so that partially missing data is retained and no warnings are given about missing data for scatterplot data in PrepareData. (Warnings are given instead from flipStandardChart about missing values). The reason for this is that the data for Scatterplots can not have additional columns attached which are used for annotations but these may or may not be used. So in many cases the warnings are not helpful.